### PR TITLE
Update Readme to include Ruby version and add 'rename' command as prereq

### PR DESCRIPTION
**Pull request by carybakker**
Minor updates to readme

### CODE REVIEW

**Comment by the-real-adammork** [comment]
Updated with some changes: 

- moved the rename dependency note to the Secrets README
- Added links to useful (public) MC documentation
- retab'd the secrets_manager scripts
- Updated the repo-structure table

@carybakker

File: scripts/secrets_manager [hunk 1]
```diff
@@ -6,34 +6,58 @@ function check_dependencies() {
...
+      echo 'rename is still not installed. installing with `brew install rename`' >&2
+      exit 1
+    fi
+  fi
+
+  echo "All dependencies installed"
 }
 
 function check_dependencies_silent() {
   # Only emits messages upon failure & exit
```
Comment (carybakker) [inline scripts/secrets_manager]: @the-real-adammork We should DRY this out and have one function that can be configured to log or not log output via a param

File: README.md [hunk 2]
```diff
@@ -52,7 +54,7 @@ The workspace can be built with `make`.
     git submodule update --init --recursive
     ```
 
-1. Install Ruby
+1. Install Ruby 3.x
```
Comment (the-real-adammork) [inline README.md:57]: why do we need ruby 3.x ? 

CICD is using 2.7, and Ive been using 2.6.x/2.7.x successfully.

**Comment by carybakker** [comment]
@the-real-adammork You should open a new PR for this branch and add me as a reviewer, since the intent of the PR has  shifted primarily to your script changes.  Thanks

**Pull request closed without merging**